### DISCUSSION TITLE
Error by trying to send an alerting to MS Teams with the formatted template

### DISCUSSION THREAD

**Discussion by rumen-vasilev** [ptarmiganlabs/butler#476] [Q&A] [answered]
Hello guy,

I am getting an error trying to send an alerting to MS Teams with the formatted template. I have just copied and linked the standard template from github without customizing it. 

Error mesage:
error: TEAMSNOTIF: Error processing Teams template file: SyntaxError: Unexpected token W in JSON at position 3955

Do you have the same problem and can you help me with it?

**Comment by rumen-vasilev** [comment]
I forgot to say that sending basic alerts to MS Teams works fine on my site.

**Reply by mountaindude** [reply]
Need a bit more info to narrow this down.


Which Butler version are you using?

What exact command is used to start Butler?

**Reply by mountaindude** [reply]
Also, is it possible to provide a copy of the errors you are seeing ?

Just copy paste it here.

**Reply by rumen-vasilev** [reply]
Hi Göran,

I run the stand-alone executable (Version 8.1.0). I start the .exe with this command in Power Shell ### **".\butler.exe -c test.yaml"**. Email-Alerting works very good by failed jobs in QMC. Basic MS Teams-Alerting without template also works. If I switch to formatted and use the standard [template ](https://github.com/ptarmiganlabs/butler/blob/master/src/config/teams_templates/failed-reload.handlebars)provided in GitHub, I get the following error by failing a job in the QMC:

 _**error: TEAMSNOTIF: Error processing Teams template file: SyntaxError: Unexpected token W in JSON at position 3955**_

Thanks for your support!

**Reply by mountaindude** [reply]
I suspect it's a matter of Butler not finding the template files.

What does the `teamsNotification.templateFile` setting in your YAML config file look like?

If you see something like `TypeError: Cannot read properties of undefined (reading 'statusText')` in your log output that's a clue that the path is incorrect. You can then try setting this to an absolute path, i.e. something like c:\path\to\failed-reload.handlebars

You can also try increasing the log level by adding `--loglevel verbose` or even `--loglevel debug` when starting Butler.
Do that and then re-create the error. Then either post the log here, starting a few lines before the error occurs until a few lines after. The important thing is to get the full context of the error, i.e. *all* log lines indicating an error.

If you don't want to post the log here you can send it to goran <at> ptarmiganlabs <dot> com and I'll take a look. 
Others have had similar problems, those times it's been issues with relative/absolute paths to the handlebars file.

**Comment by mountaindude** [comment]
Scratch the idea around template files not being found. 

More likely that there's an issue involving the library used to parse the template file into the message that's then sent to Teams.
The library has been updated to a new version and it could very well be that it works slightly differently when it comes to handling of characters like (for example) back-slash \

I'll look into that.

**Reply by rumen-vasilev** [reply]
Hi Göran,

Maybe you are right with the back-slash. I actually use it in the path. Please find below snapshots from config and log file.
Snapshot from config file:
![image](https://user-images.githubusercontent.com/83919259/172832582-6e182549-fb91-43cf-b7cc-ad3f3003ccb5.png)

log (degug level) after failing the QMC job:
![image](https://user-images.githubusercontent.com/83919259/172831183-f66b8113-e89f-4cb7-a3fb-11d707fc6bce.png)

BR,
Rumen

**Reply by mountaindude** [reply]
I  can confirm that back slashes are NOT handled well by the formatted Slack and Teams messages.
They used to work fine, something has probably changed in the libraries used to render those messages.

Anyways, I have identified the problem and created a fix for it. 
Will be part of the 8.2.0 release which is due out in a day or two.

Thanks for reporting the issue!

**Comment by mountaindude** [comment, marked as answer]
@rumen-vasilev 
This bug is now tracked in #485.

The plan is to have a fix included in 8.2.0, which is due to be released in a couple of days.
